clic/include/core/cleImage.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef __CORE_CLEIMAGE_HPP
+#define __CORE_CLEIMAGE_HPP
+
+#include "cleBackend.hpp"
+#include "cleLightObject.hpp"
+#include "cleTypes.hpp"
+
+
+namespace cle
+{
+
+class Image : public LightObject
+{
+public:
+  Image() = default;
+  ~Image() override = default;
+  Image(const Image & obj) = default;
+  Image(Image && obj) = default;
+  auto
+  operator=(const Image & obj) -> Image & = default;
+  auto
+  operator=(Image && obj) -> Image & = default;
+
+  Image(const ProcessorPointer & device,
+        const cl::Memory &       data,
+        const ShapeArray &       shape,
+        const DataType &         data_type,
+        const ObjectType &       object_type,
+        const ChannelType &      channels_type);
+
+  auto
+  Fill(const float & value) const -> void;
+  auto
+  CopyDataTo(const Image & dst_obj) const -> void;
+  [[nodiscard]] auto
+  Get() const -> cl::Memory;
+  [[nodiscard]] auto
+  Device() const -> ProcessorPointer;
+  [[nodiscard]] auto
+  Bytes() const -> size_t override;
+  [[nodiscard]] auto
+  Ndim() const -> unsigned int override;
+  [[nodiscard]] auto
+  Shape() const -> ShapeArray override;
+  [[nodiscard]] auto
+  ShapeZYX() const -> ShapeArray;
+  [[nodiscard]] auto
+  Origin() const -> ShapeArray;
+  [[nodiscard]] auto
+  ObjectInfo() const -> std::string override;
+  [[nodiscard]] auto
+  DataInfo() const -> std::string override;
+  [[nodiscard]] auto
+  DataInfoShort() const -> std::string override;
+  [[nodiscard]] auto
+  IsBuffer() const -> bool;
+  [[nodiscard]] auto
+  IsImage() const -> bool;
+  [[nodiscard]] auto
+  ToString() const -> std::string override;
+  [[nodiscard]] auto
+  Data() const -> DataType override;
```

Comment:
Might it makes sense to call this `DataType` and not `Data`? I just hit this while updating code...

Reply:
agreed, renaming methods to `DataType` and `ObjectType` is planned
return type will be rename `DataTypeEnum` etc.

Reply:
Also, I moved methods to the abstract class `LightObject` to reduce code replication and increase class clarity